### PR TITLE
Add PKCS#11 Compliant Signer

### PR DESCRIPTION
**Pull request by malnick**
Thank you for sending a pull request :heart:

### Checklist

Please review the [guidelines for contributing code](../CONTRIBUTING.md#contribute-code) to this repository. And check `[x]` all the boxes that apply:

- [ ] This request **pulls a feature/bugfix branch** (right side) from my fork. *Not my master.*
- [x] This request **pulls against** `ockam-network/ockam`’s **develop branch** (left side).
- [x] Build succeeds `./build` with no linter warnings or test failures.
- [x] All code matched our code formatting [conventions](../CONTRIBUTING.md#code-format).
- [x] Commit messages match our [conventions](../CONTRIBUTING.md#commit-messages) and all commits
are [signed](#signed-commits).
- [x] This request includes tests and documentation for all new code.

### Proposed Changes
Please describe the changes in your pull request.

If this pull request resolves an already recorded bug or a feature request, be sure to link to that issue.

### CODE REVIEW

**Comment by mrinalwadhwa** [comment]
@malnick could you change those `ci:` commit tags to `build:`
we may need to clarify descriptions here https://github.com/ockam-network/ockam/blob/master/commitlint.config.js

File: key/pkcs11/publickey.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+// PKCS11PublicKey is an implementation of an ockam.PublicKey
+// interface for the pkcs11 package
+type PKCS11PublicKey struct {
+	label     string
+	owner     ockam.Entity
+	publicKey crypto.PublicKey
+}
+
+// Label is a getting for the private lable attribute
```
Comment (mrinalwadhwa) [inline key/pkcs11/publickey.go]: typo `lable`

**Comment by mrinalwadhwa** [comment]
@malnick once you make the minor teaks above we're ready to merge this 👍 Thank you

File: key/pkcs11/publickey.go (hunk 1 above)
Comment (malnick) [reply key/pkcs11/publickey.go]: fixed.

**Pull request merged**